src/TestExplorer/TestExplorer.ts

```diff
@@ -80,11 +83,28 @@ export class TestExplorer {
             switch (event) {
                 case FolderEvent.add:
                     if (folder) {
-                        folder.addTestExplorer();
-                        // discover tests in workspace but only if disableAutoResolve is not on.
-                        // discover tests will kick off a resolve if required
-                        if (!configuration.folder(folder.workspaceFolder).disableAutoResolve) {
-                            folder?.testExplorer?.discoverTestsInWorkspace();
+                        if (folder.swiftPackage.getTargets("test").length > 0) {
+                            folder.addTestExplorer();
+                            // discover tests in workspace but only if disableAutoResolve is not on.
+                            // discover tests will kick off a resolve if required
+                            if (!configuration.folder(folder.workspaceFolder).disableAutoResolve) {
+                                folder.testExplorer?.discoverTestsInWorkspace();
+                            }
+                        }
+                    }
+                    break;
+                case FolderEvent.packageUpdated:
+                    if (folder) {
+                        const testTargets = folder.swiftPackage.getTargets("test");
```

Comment:
Suggestion: `const hasTestTargets = folder.swiftPackage.getTargets("test").length > 0`, since the rest of the code doesn't need the array.

Reply:
Not sure it'll make a difference to performance but it make the code clearer so I've implemented it